### PR TITLE
8256253: Defer Biased Locking obsoletion to JDK 18

### PR DESCRIPTION
**Pull request by pchilano**
Hi all,

Please review the following change to defer biased-locking obsoletion to 18, as discussed in [1].
I updated the special_jvm_flags table and re-enabled gtest verify_special_flags.

Thanks,
Patricio

[1] https://mail.openjdk.java.net/pipermail/hotspot-dev/2020-November/045454.html

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Testing

|     | Linux aarch64 | Linux arm | Linux ppc64le | Linux s390x | Linux x64 | Linux x86 | Windows x64 | macOS x64 |
| --- | ----- | ----- | ----- | ----- | ----- | ----- | ----- | ----- |
| Build | ✔️ (1/1 passed) | ✔️ (1/1 passed) | ✔️ (1/1 passed) | ✔️ (1/1 passed) | ✔️ (6/6 passed) | ✔️ (2/2 passed) | ✔️ (2/2 passed) | ✔️ (2/2 passed) |
| Test (tier1) |    |     |     |     |  ✔️ (9/9 passed) | ❌ (1/9 failed) | ✔️ (9/9 passed) | ✔️ (9/9 passed) |

**Failed test task**
- [Linux x86 (hs/tier1 compiler)](https://github.com/pchilano/jdk/runs/1418982414)

### Issue
 * [JDK-8256253](https://bugs.openjdk.java.net/browse/JDK-8256253): Defer Biased Locking obsoletion to JDK 18

### Reviewers
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Coleen Phillimore](https://openjdk.java.net/census#coleenp) (@coleenp - **Reviewer**) ⚠️ Review applies to b949dcd36edd33cff8a8727d68a048dd5e2bb5f7
 * [Daniel D. Daugherty](https://openjdk.java.net/census#dcubed) (@dcubed-ojdk - **Reviewer**)
 * [Martin Doerr](https://openjdk.java.net/census#mdoerr) (@TheRealMDoerr - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1281/head:pull/1281`
`$ git checkout pull/1281`

### CODE REVIEW

File: src/hotspot/share/runtime/arguments.cpp [hunk 1]
```diff
@@ -521,14 +521,14 @@ static SpecialFlag const special_jvm_flags[] = {
...
-  { "PrintPreciseBiasedLockingStatistics", JDK_Version::jdk(15), JDK_Version::jdk(16), JDK_Version::jdk(17) },
+  { "UseBiasedLocking",             JDK_Version::jdk(15), JDK_Version::jdk(18), JDK_Version::jdk(19) },
+  { "BiasedLockingStartupDelay",    JDK_Version::jdk(15), JDK_Version::jdk(18), JDK_Version::jdk(19) },
+  { "PrintBiasedLockingStatistics", JDK_Version::jdk(15), JDK_Version::jdk(18), JDK_Version::jdk(19) },
+  { "BiasedLockingBulkRebiasThreshold",    JDK_Version::jdk(15), JDK_Version::jdk(18), JDK_Version::jdk(19) },
+  { "BiasedLockingBulkRevokeThreshold",    JDK_Version::jdk(15), JDK_Version::jdk(18), JDK_Version::jdk(19) },
+  { "BiasedLockingDecayTime",              JDK_Version::jdk(15), JDK_Version::jdk(18), JDK_Version::jdk(19) },
+  { "UseOptoBiasInlining",                 JDK_Version::jdk(15), JDK_Version::jdk(18), JDK_Version::jdk(19) },
+  { "PrintPreciseBiasedLockingStatistics", JDK_Version::jdk(15), JDK_Version::jdk(18), JDK_Version::jdk(19) },
   { "CriticalJNINatives",                  JDK_Version::jdk(16), JDK_Version::jdk(17), JDK_Version::jdk(18) },
```
Comment (dholmes-ora) [inline src/hotspot/share/runtime/arguments.cpp:532]: To maintain the table ordering this entry needs to move above the BL entries now.

**Review by dholmes-ora — APPROVED** [review]
Hi Patricio,

One minor issue - see below.

Thanks,
David

File: src/hotspot/share/runtime/arguments.cpp (hunk 1 above)
Comment (coleenp) [reply src/hotspot/share/runtime/arguments.cpp:532]: So these are ordered by the second field?

**Review by coleenp — APPROVED** [review]
Looks good.  Amazing how many product options there are.

**Comment by coleenp** [comment]
I have an alternate idea.  I wonder if for LTS we should remove the BiasedLocking other options as planned, and leave the main option in place until JDK 18.  Then we wouldn't have to test and support the variations of the code that the other options turn on and off.

**Comment by pchilano** [comment]
> Hi Patricio,
> 
> One minor issue - see below.
> 
> Thanks,
> David
Hi David,

Thanks for the review!

File: src/hotspot/share/runtime/arguments.cpp (hunk 1 above)
Comment (pchilano) [reply src/hotspot/share/runtime/arguments.cpp]: Fixed.

**Comment by pchilano** [comment]
> Looks good. Amazing how many product options there are.
Hi Coleen,
Thanks for the review!

Patricio

**Review by dcubed-ojdk — APPROVED** [review]
Thumbs up.

**Comment by pchilano** [comment]
> Thumbs up.
Thanks Dan!

Patricio

**Review by TheRealMDoerr — APPROVED** [review]
Thanks for doing this!

**Review by dholmes-ora — APPROVED** [review]

**Comment by pchilano** [comment]
@TheRealMDoerr thanks for the review!

**Pull request closed without merging**